### PR TITLE
Respect  JULIA_CPU_TARGET set in Make.inc when building pkgimgs

### PR DESCRIPTION
**Pull request by gbaraldi**

### CODE REVIEW

**Comment by vchuravy** [comment]
This should already work... at least I tested this, but using the environment variable since this use of `-C` isn't really they way it's meant to be used.

**Comment by gbaraldi** [comment]
It's how we do it when building the sysimg, and I found this causing some issues if the envar isn't set but Make.inc is

**Comment by vchuravy** [comment]
I guess after https://github.com/JuliaLang/julia/pull/48809 this is fine, but I don't think we ever answered the question: What's the JIT target going to be if you provide a multi-versioning string as the target...

**Comment by staticfloat** [comment]
```
More than one command line CPU targets specified without a `--output-` flag specified
```

**Comment by vchuravy** [comment]
Yes that's what I would have expected

File: pkgimage.mk [hunk 1]
```diff
@@ -30,14 +30,14 @@ all-debug:   $(addprefix cache-debug-, $(STDLIBS))
 define stdlib_builder
 ifneq ($(filter $(1),$(INDEPENDENT_STDLIBS)),)
 $$(BUILDDIR)/stdlib/$1.release.image: $$($1_SRCS) $$(addsuffix .release.image,$$(addprefix $$(BUILDDIR)/stdlib/,$2)) $(build_private_libdir)/sys.$(SHLIB_EXT)
-	@$$(call PRINT_JULIA, $$(call spawn,$$(JULIA_EXECUTABLE)) --startup-file=no --check-bounds=yes -e 'Base.compilecache(Base.identify_package("$1"))')
+	@$$(call PRINT_JULIA, JULIA_CPU_TARGET=$(JULIA_CPU_TARGET) $$(call spawn,$$(JULIA_EXECUTABLE)) --startup-file=no --check-bounds=yes -e 'Base.compilecache(Base.identify_package("$1"))')
```
Comment (staticfloat) [inline pkgimage.mk:33]:
```suggestion
	@JULIA_CPU_TARGET=$(JULIA_CPU_TARGET) $$(call PRINT_JULIA, $$(call spawn,$$(JULIA_EXECUTABLE)) --startup-file=no --check-bounds=yes -e 'Base.compilecache(Base.identify_package("$1"))')
```

**Comment by staticfloat** [comment]
Testing locally shows that this works now; this functionality is not tested on CI, as [we set the environment variables already](https://github.com/JuliaCI/julia-buildkite/blob/0e1d450748e7c67cf8a95df28e9ab40761c8c6de/utilities/build_envs.sh#L86).

**Pull request merged**